### PR TITLE
add optional field: ResourceRequests in the swagger

### PR DESCRIPTION
**Pull request by neuqlz**

### CODE REVIEW

File: specification/appplatform/resource-manager/Microsoft.AppPlatform/preview/2020-11-01-preview/appplatform.json [hunk 1]
```diff
@@ -3925,16 +3925,20 @@
       "properties": {
         "cpu": {
           "format": "int32",
-          "description": "Required CPU, basic tier should be 1, standard tier should be in range (1, 4)",
+          "description": "Required CPU, basic tier should be 1, standard tier should be in range (1, 4), it will be deprecated since api version 2020-11-01-preview, please leverage the ResourceRequest field to set the cpu size.",
```
Comment (allxiao) [inline specification/appplatform/resource-manager/Microsoft.AppPlatform/preview/2020-11-01-preview/appplatform.json:3928]: Not specifically for this PR.

* I think we should use `[1, 4]` here. 
* `,` should be replaced with `.`
* Inconsistent case: `api` -> `API`, `cpu` -> `CPU`.
* Some grammar error.

Maybe:
```
Required CPU. This should be 1 for Basic tier, and in range [1, 4] for Standard tier. This is deprecated starting from API version 2020-11-01-preview. Please use the ResourceRequest field to set the CPU size.
```

File: specification/appplatform/resource-manager/Microsoft.AppPlatform/preview/2020-11-01-preview/appplatform.json [hunk 2]
```diff
@@ -3925,16 +3925,20 @@
...
         "memoryInGB": {
           "format": "int32",
-          "description": "Required Memory size in GB, basic tier should be in range (1, 2), standard tier should be in range (1, 8)",
+          "description": "Required Memory size in GB, basic tier should be in range (1, 2), standard tier should be in range (1, 8), it will be deprecated since api version 2020-11-01-preview, please leverage the ResourceRequest field to set the memory size.",
           "default": 1,
           "type": "integer"
         },
+        "resourceRequest": {
+          "$ref": "#/definitions/ResourceRequest",
+          "description": "The request of quantity for cpu and memory size."
```
Comment (allxiao) [inline specification/appplatform/resource-manager/Microsoft.AppPlatform/preview/2020-11-01-preview/appplatform.json:3940]: The requested CPU and memory quantity.

File: specification/appplatform/resource-manager/Microsoft.AppPlatform/preview/2020-11-01-preview/appplatform.json (hunk 1 above)
Comment (allxiao) [reply specification/appplatform/resource-manager/Microsoft.AppPlatform/preview/2020-11-01-preview/appplatform.json:3928]: Same for below.

File: specification/appplatform/resource-manager/Microsoft.AppPlatform/preview/2020-11-01-preview/appplatform.json [hunk 3]
```diff
@@ -4014,6 +4018,20 @@
         }
       }
     },
+    "ResourceRequest": {
+      "description": "Deployment resource request payload",
+      "type": "object",
+      "properties": {
+        "cpu": {
+          "description": "Required CPU, 1c can be represent by 1 or 1000m, basic tier should be 500m or 1, standard tier should be 500m or the value in range (1, 4).",
```
Comment (allxiao) [inline specification/appplatform/resource-manager/Microsoft.AppPlatform/preview/2020-11-01-preview/appplatform.json:4026]: Maybe:

```
Required CPU. 1 core can be represented by 1 or 1000m. This should be 500m or 1 for Basic tier, and {500m, 1, 2, 3, 4} for Standard tier.
```

File: specification/appplatform/resource-manager/Microsoft.AppPlatform/preview/2020-11-01-preview/appplatform.json [hunk 4]
```diff
@@ -4014,6 +4018,20 @@
...
+          "type": "string"
+        },
+        "memory": {
+          "description": "Required Memory size, 1GB can be represent by 1Gi or 1024Mi, basic tier should be 500m or the value in range (1, 2), standard tier should be 500m or the value in range (1, 8).",
```
Comment (allxiao) [inline specification/appplatform/resource-manager/Microsoft.AppPlatform/preview/2020-11-01-preview/appplatform.json:4030]: Maybe

```
Required memory. 1 GB can be represented by 1Gi or 1024Mi. This should be {500m, 1Gi, 2Gi} for Basic tier, and {500m, 1Gi, 2Gi, ..., 8Gi} for Standard tier.
```

**Review by allxiao — CHANGES_REQUESTED** [review]
We need to update the descriptions.

File: specification/appplatform/resource-manager/Microsoft.AppPlatform/preview/2020-11-01-preview/appplatform.json (hunk 2 above)
Comment (neuqlz) [reply specification/appplatform/resource-manager/Microsoft.AppPlatform/preview/2020-11-01-preview/appplatform.json]: fixed

File: specification/appplatform/resource-manager/Microsoft.AppPlatform/preview/2020-11-01-preview/appplatform.json (hunk 3 above)
Comment (neuqlz) [reply specification/appplatform/resource-manager/Microsoft.AppPlatform/preview/2020-11-01-preview/appplatform.json]: fixed

File: specification/appplatform/resource-manager/Microsoft.AppPlatform/preview/2020-11-01-preview/appplatform.json (hunk 4 above)
Comment (neuqlz) [reply specification/appplatform/resource-manager/Microsoft.AppPlatform/preview/2020-11-01-preview/appplatform.json]: fixed

File: specification/appplatform/resource-manager/Microsoft.AppPlatform/preview/2020-11-01-preview/appplatform.json (hunk 1 above)
Comment (neuqlz) [reply specification/appplatform/resource-manager/Microsoft.AppPlatform/preview/2020-11-01-preview/appplatform.json]: fixed

File: specification/appplatform/resource-manager/Microsoft.AppPlatform/preview/2020-11-01-preview/appplatform.json [hunk 5]
```diff
@@ -4014,6 +4018,20 @@
         }
       }
     },
+    "ResourceRequest": {
```
[the remark this answers is not in the document]
Comment (neuqlz) [reply specification/appplatform/resource-manager/Microsoft.AppPlatform/preview/2020-11-01-preview/appplatform.json]: yep, fixed.

**Comment by allxiao** [comment]
lgtm

**Pull request merged**